### ISSUE TITLE
Installation does not properly take into account sub-dependencies

### ISSUE THREAD

**Issue by tizz98** [enpaul/tox-poetry-installer#41]
Reproduce repo: https://github.com/tizz98/tox-poetry-installer-bug-repro

## How to reproduce

_Ensure you have `poetry==1.1.4` installed_

```bash
git clone git@github.com:tizz98/tox-poetry-installer-bug-repro.git
cd tox-poetry-installer-bug-repro

poetry --version  # 1.1.4
pyenv virtualenv 3.8.6 tpibr
pyenv local tpibr

poetry install
tox
```

### Expected result

Pytest should run the tests.

```
==================================== test session starts ====================================
platform linux -- Python 3.8.6, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/noteable/dev/tizz98/tox-poetry-installer-bug-repro
collected 1 item                                                                            

test_foo.py .                                                                         [100%]

===================================== 1 passed in 0.01s =====================================
```

### Error

```
    ModuleNotFoundError: No module named 'ipython_genutils'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-60h8tk8n/ipypandex_68cec70a49ca4cd7b86c78b3788bf74a/setup.py", line 43, in <module>
        setup(
      File "/home/noteable/.pyenv/versions/3.8.6/lib/python3.8/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/home/noteable/.pyenv/versions/3.8.6/lib/python3.8/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/home/noteable/.pyenv/versions/3.8.6/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-60h8tk8n/ipypandex_68cec70a49ca4cd7b86c78b3788bf74a/setup.py", line 40, in run
        raise RuntimeError("ipypandex requires an ipython installation to work")
    RuntimeError: ipypandex requires an ipython installation to work
```

or 

```
    ModuleNotFoundError: No module named 'IPython'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-m4usifx2/ipypandex_77e6e67871334c719f775eaa405b2ecb/setup.py", line 43, in <module>
        setup(
      File "/home/noteable/.pyenv/versions/3.8.6/lib/python3.8/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/home/noteable/.pyenv/versions/3.8.6/lib/python3.8/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/home/noteable/.pyenv/versions/3.8.6/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-m4usifx2/ipypandex_77e6e67871334c719f775eaa405b2ecb/setup.py", line 40, in run
        raise RuntimeError("ipypandex requires an ipython installation to work")
    RuntimeError: ipypandex requires an ipython installation to work
```

### Explanation

The `ipython` package is a sub-dependency of `ipypandex` and when installing with `poetry` regularly it works fine because sub-dependencies are installed first. In this case though, the `tox-poetry-installer` plugin is just installing dependencies in a semi-random order which doesn't always work.

Even when specifying `ipython` explicitly as a dev dependency the same issue occurs.

### Proposed fix

`tox-poetry-installer` should use a similar approach as how the `poetry` cli installs packages, see https://github.com/python-poetry/poetry/blob/18a9e2dfad36332c9b6c53f7a1cf980b66325fe4/poetry/installation/installer.py#L223

**Comment by enpaul** [comment]
Thank you for the very comprehensive bug report and reproducer! I'll definitely work on getting this fixed.

After taking a quick look at the reproducing repository you linked, I've found a work around that also informs what may be happening under the hood. By setting `ipypandex` as an explicit item in `locked_deps` rather than relying on the `install_dev_deps` option the installation worked as expected. This is the diff on the `tox.ini`:

```
@@ -11,6 +11,7 @@ passenv = *
 basepython =
     py38: python3.8
 commands = pytest -vv --maxfail=2  -W always {posargs}
-install_dev_deps = true
+locked_deps =
+    ipypandex
 requires =
     tox-poetry-installer[poetry] == 0.6.2
```

The above may serve as a workaround to get you unblocked in the short term.

In the long term however you're absolutely correct that this is broken behavior. The fact that it works with `locked_deps` and not with `install_dev_deps` indicates that there may be an issue with the way that the plugin interfaces with Poetry's locker class to determine what is and isn't a dev-dependency. I'll take another look at how the locker does resolution and work on a permanent resolution to this problem. 

Thank you again for the bug report and for trying out the project!

**Comment by enpaul** [comment]
@tizz98 I just uploaded version 0.6.3 which should fix this bug (the updated version worked as expected using your test repo, thank you very much for providing that!) When you have a chance could you update your version of the plugin to see if the problem is solved?

**Comment by tizz98** [comment]
Yep, this issue is fixed, thanks! Although I ran into another unrelated issue, so I'll create a new issue for that. Thanks for the responsiveness :pray:
